core/src/consensus/stake/mod.rs

```diff
@@ -168,6 +204,39 @@ pub fn get_stakes(state: &TopLevelState) -> StateResult<HashMap<Address, u64>> {
     Ok(result)
 }
 
+fn change_params(
+    state: &mut TopLevelState,
+    metadata_seq: usize,
+    params: CommonParams,
+    signatures: &[Signature],
+) -> StateResult<()> {
+    state.update_params(metadata_seq, params.clone())?;
```

Comment:
Why do we update before the signature check?

Reply:
Because checking signatures are the most expensive work.

Reply:
`update_params` doesn't short-circuit so it doesn't change the average cost.

Reply:
It does when the metadata_seq is different.
The code that returns `Ok(())` should be replaced to an error. Anyway, it does.

Reply:
Okay then, can we mention the purpose of the unusual code the placement?

Reply:
done